sled-agent/config-reconciler/src/reconciler_task/datasets.rs

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Module for Omicron datasets.
+//!
+//! There is no separate tokio task here; our parent reconciler task owns this
+//! set of datasets and is able to mutate it in place during reconciliation.
```

Comment:
I'm trying to make this comment "There is no separate tokio task here" make sense, given that there is a `DatasetTaskHandle` field in the `OmicronDatasets` field below.

Seems like there is **both** a task-independent dataset structure, as well as a task?

Reply:
Mm yeah this is confusing. My intent was "this datastructure isn't one that spawns a task and uses a channel to communicate to it", which is true, but it does need to use the dataset serialization task handle that's spawned elsewhere (and is also used to service support bundle nested dataset operations and inventory collections).

Maybe `DatasetSerializationTask` should be `ZfsSerializationTask` - would that be clearer?

Reply:
I reworded the comment in [cf76de8](https://github.com/oxidecomputer/omicron/pull/8218/commits/cf76de8537d8f53b20b12d6960d9e49734e188dc)